x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -135,6 +135,12 @@ export interface ActionTypeModel<ActionConfig = any, ActionSecrets = any, Action
     >
   > | null;
   actionParamsFields: React.LazyExoticComponent<ComponentType<ActionParamsProps<ActionParams>>>;
+  customConnectorSelectItem?: {
```

Comment:
@YulNaumenko I split this up into two different operations because we only display the icon (aka `getComponent`) after the row has been selected, but we display the text in the dropdown of the combo box. Let me know if you have a better idea. I'm happy to change it.

Reply:
Thank you for the explanation. I think we better create the proper reusable interface instead of providing that details coupled to the ActionTypeModel. But the rest looks good to me.